packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -376,6 +376,10 @@ export const ComboBoxMixin = (subclass) =>
         }
 
         this.__restoreFocusOnClose = true;
+
+        if (!this.filter) {
```

Comment:
The `filter` property can be an empty string either when the user fills in an empty filter query or after a value has been selected from the dropdown. In the latter case, the empty string is needed to show the full non-filtered list of items in case the user opens the dropdown again.

That being so, it becomes impossible to determine whether `this.filter === ''` is caused by changing the filter query or selecting a value in the past. That currently causes the value item to get focused when clearing the filter query in the closed state while doing the same in the open state leads to no item being focused.

https://user-images.githubusercontent.com/5039436/159862542-6f5f2e5b-3963-4dcc-9d04-df5a85f4a763.mov

Given that the inconsistency has existed since even before the PR that caused the regression and nobody has complained, I think it is not a big problem after all.

Reply:
I found a better solution that fixes this inconsistency along the way. See https://github.com/vaadin/web-components/commit/d3e81408a93cacd7f5c54bb2821842566ecbaec2.

Reply:
Here is the unit test for this case:

https://github.com/vaadin/web-components/commit/d3e81408a93cacd7f5c54bb2821842566ecbaec2#diff-95666241933a0971a1c2958cbd8143ccfbd4b269d293ab5b42180fe539c985b7R211-R215